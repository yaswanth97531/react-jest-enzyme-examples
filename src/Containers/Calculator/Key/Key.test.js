import Key from './Key';

let wrapper;

beforeEach(() => {
  wrapper = shallow( 
    <Key
      keyAction={jest.fn()}
      keyType={''}
      keyValue={''}
    /> );
});

describe('<Key /> rendering', () => {
  it('should render correctly', () => {
    const tree = renderer.create( <Key / > ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a <div>', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the value of keyValue', () => {
    wrapper.setProps({ keyValue: 'test' });
    expect(wrapper.text()).toEqual('test');
  });
});

describe('<Key /> interactions', () => {});

describe('<Key /> lifecycle method invocations', () => {});