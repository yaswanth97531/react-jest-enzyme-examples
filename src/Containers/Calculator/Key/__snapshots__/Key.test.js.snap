// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Key /> rendering should render correctly 1`] = `
<div
  className="key-container undefined"
  onClick={[Function]}
>
  <p
    className="key-value"
  />
</div>
`;
